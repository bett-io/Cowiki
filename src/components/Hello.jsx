import React from 'react';

const Hello = () => (
  <div>
    <div>hello world</div>
    <a href="/about">about</a>
  </div>
);

export default Hello;
