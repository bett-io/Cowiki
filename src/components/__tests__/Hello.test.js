/* eslint-disable react/jsx-filename-extension */

import React from 'react';

import toJson from 'enzyme-to-json';
import Hello from '../Hello';
import { shallow } from 'enzyme';

describe('Hello component', () => {
  test('should pass snapshot check', () => {
    const rendered = shallow(<Hello />);
    expect(toJson(rendered)).toMatchSnapshot();
  });

  test('should render "hello world" string', () => {
    expect(shallow(<Hello />).contains(<div>hello world</div>)).toBeTruthy();
  });

  test('should contain a link to /about', () => {
    expect(shallow(<Hello />).contains(<a href="/about">about</a>)).toBeTruthy();
  });
});
