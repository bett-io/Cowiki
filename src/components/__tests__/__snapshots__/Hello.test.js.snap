// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component should pass snapshot check 1`] = `
<div>
  <div>
    hello world
  </div>
  <a
    href="/about"
  >
    about
  </a>
</div>
`;
